cypress/e2e/right-panel/pinned-messages-card.spec.ts

```diff
@@ -0,0 +1,209 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/// <reference types="cypress" />
+
+import { HomeserverInstance } from "../../plugins/utils/homeserver";
+
+const ROOM_NAME = "Test room";
+
+describe("PinnedMessagesCard", () => {
+    let homeserver: HomeserverInstance;
+
+    /**
+     * Send a message and pin it.
+     * @param message Message to send and pin
+     */
+    const sendAndPin = (message: string) => {
+        // Send a message
+        cy.getComposer().type(`${message}{enter}`);
+
+        // Assert that the event tile is rendered
+        cy.contains(".mx_EventTile_last", `${message}`);
+
+        // Hover the event tile and click the options button
+        cy.get(".mx_EventTile_last").realHover().findByRole("button", { name: "Options" }).click();
+
+        // Click the pin button
+        cy.findByRole("menuitem", { name: "Pin" }).should("be.visible").click();
+    };
+
+    // Exclude timestamps from snapshots
+    const percyCSS = ".mx_MessageTimestamp { visibility: hidden !important; }";
+
+    beforeEach(() => {
+        cy.startHomeserver("default").then((data) => {
+            homeserver = data;
+            cy.initTestUser(homeserver, "Hanako").then(() => {
+                cy.createRoom({ name: ROOM_NAME });
+            });
+        });
+
+        // Enable message pinning feature
+        cy.enableLabsFeature("feature_pinning");
+
+        cy.viewRoomByName(ROOM_NAME);
+
+        // Send and pin a message
+        sendAndPin("First pinned message");
+    });
+
+    afterEach(() => {
+        cy.stopHomeserver(homeserver);
+    });
+
+    it("should update when messages are pinned", () => {
+        // Assert that the button to display the pinned message card is rendered inside the room header, and click it
+        cy.get(".mx_RoomHeader").within(() => {
+            cy.findByRole("button", { name: "Pinned messages" }).should("be.visible").click();
+        });
+
+        // Send and pin the second message
+        sendAndPin("The second one");
+
+        // Assert that the pinned message card is rendered
+        cy.get(".mx_PinnedMessagesCard").within(() => {
+            // Assert that the pinned messages are rendered
+            cy.get(".mx_PinnedEventTile").should("have.length", 2);
+        });
+
+        // Skip taking the snapshot to save quota usage
+        //
+        // Take a snapshot of RightPanel
+        // cy.get(".mx_RightPanel").percySnapshotElement("Pinned messages card - pinned messages", {
+        //     percyCSS,
+        //     widths: [264], // Emulate the UI. The value is based on minWidth specified on MainSplit.tsx
+        // });
```

Comment:
I'd suggest removing these altogether. It's just clutter, and if somebody wants to add more snapshots later it's pretty easy to figure out.